services/QuillLMS/app/lib/utils/string.rb

```diff
@@ -10,6 +10,8 @@ def self.last_name_or_name(first_name_last_name_string)
     last || first
   end
 
+  def self.to_username(str) = str.downcase.strip.gsub(/\s/, '.')
```

Comment:
I think this method belongs in a LearnWorlds namespace.  app/lib suggests this is a general method that applies to all strings.  I don't think there's a generally accepted to_username method.